lib/netext/httpext/request.go

```diff
@@ -116,6 +134,56 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 		Headers: preq.Req.Header,
 	}
 	if preq.Body != nil {
+		if len(preq.Compressions) > 0 {
+			var contentEncoding string
+			var prevBuf = preq.Req.Body
+			var buf *bytes.Buffer
+			for _, compressionType := range preq.Compressions {
+				if buf != nil {
+					prevBuf = ioutil.NopCloser(buf)
+				}
+				buf = new(bytes.Buffer)
+
+				if contentEncoding != "" {
+					contentEncoding += ", "
+				}
+				contentEncoding += compressionType.String()
+				var w io.WriteCloser
+				switch compressionType {
+				case CompressionTypeGzip:
+					w = gzip.NewWriter(buf)
+				case CompressionTypeDeflate:
+					// Have the compression level as an option somewhere ?
+					// the error is only possible if we provide value outside of the range [-2, 9]
+					w, _ = flate.NewWriter(buf, flate.DefaultCompression)
+				default:
+					return nil, fmt.Errorf("unknown compressionType %s", compressionType)
+				}
+				defer func() { _ = w.Close() }()
+				var _, err = io.Copy(w, prevBuf)
+				if err != nil {
+					return nil, err
+				}
+				if err = w.Close(); err != nil {
+					return nil, err
+				}
+			}
+
+			preq.Req.Body = ioutil.NopCloser(buf)
+			if preq.Req.Header.Get("Content-Length") == "" {
+				preq.Req.ContentLength = int64(buf.Len())
+			} else {
+				// TODO: print line
```

Comment:
I think, if we can't print the script line, a nice compromise would be to print the actual request URL. Something like:
> Both compression and the Content-Length header were specified in the METHOD request for 'URL', the custom header has precedence and won't be overwritten. This will likely result in invalid data being sent to the server.

Reply:
Oh ... having different messages for when they have specified only one of them and then when they have specified both seems to me a little bit too excessive :). Message per each header is okay :)

Reply:
Not sure I understand what you're trying to say, sorry

Reply:
I intend on having two seperate log messages one for each header that I might overwrite as it currently is. 
Adding the URL seems good, and I suppose it will help somewhat :)

Reply:
There was a misunderstanding it seems :smile:  I was fine with having 2 (or rather 3, since there's one for just having a `body` and predefined `Content-Length` header as well) separate warning messages. I just gave the `Content-Length` above since I was commenting on this particular `TODO` :wink: 
For the `TODO` on line 188 it can be something like:
> Both compression and the Content-Encoding header were specified in the <METHOD> request for '<URL>', the custom header has precedence and won't be overwritten. This will likely result in invalid data being sent to the server.

Reply:
Actually, I'm not sure that we need to show a warning message for `Content-Length` here (when handling `compression`) at all. Since if there's compression, then there's a body, ergo the user will **always** see both error messages and the body one is a bit more general than this, so maybe only it should stay?
(the `Content-Encoding` should remain here of course, since it doesn't have a duplicate above)

Reply:
Yeah but they could've only said 
```
http.post("https://example.com", "123456789", {compression: "gzip", headers:{"Content-Length": 9}})
```
at which point this code will not overwrite `Content-Length` but it will be not `9`

Reply:
That was my problem ... there will be two messages. 
Do you think we should not show a message if the lengths match ?

Reply:
I want to say no, since I think we should simply always show only the more general warning message, i.e. if there's a `body` and `Content-Length` was manually set - regardless of whether there's any compression... Something like:
> For METHOD requests, the Content-Length header is automatically calculated from the length of the supplied body and any compression. Specifying it manually for 'URL' will overwrite that and will likely result in invalid data being sent to the server.

but... I have the feeling that scripts generated by the HAR converter will result in a LOT of warnings :man_facepalming: That's why I [suggested](https://github.com/loadimpact/k6/pull/989#issuecomment-481533499) we do the header validation in a future PR :wink:  

So... maybe for now we should disable the general warning and leave only the two warnings for when `compression` is enabled?